sdk/core/core/inc/_az_cfg.h

```diff
@@ -14,8 +14,9 @@
 
 #ifdef _MSC_VER
 
-// warning C4710: '...': function not inlined
-#pragma warning(error : 4710)
+// warning C4061: enumerator 'AZ_OK' in switch of enum 'az_result' is not explicitly handled by a
+// case label
+#pragma warning(disable : 4061)
```

Comment:
Should we fix this in code (maybe by using if/else) instead of disabling the warning? We probably want to make sure that all enum cases are there within a switch/case statement.

Reply:
It does not make sense, and it will be impossible to support.
For instance, if you have an `enum az_error { AZ_OK, AZ_OVERFLOW, AZ_IOT_MQTT, ... 100 more... }`, it would require you to mention _every_ error _every_ switch:
```
switch (error)
{
case AZ_OK: ...
default: // no longer works, you have to handle all 100 errors here
}
```

Reply:
So basically it will be almost impossible to write any switch/case statement, and for no good reason.